production/jaeger-production-template.yml

```diff
@@ -0,0 +1,345 @@
+          - "-cassandra.keyspace=${KEYSPACE}"
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-collector
+    labels:
+      app: jaeger
+      jaeger-infra: collector-service
+  spec:
+    ports:
+    - name: jaeger-collector
+      port: 14267
+      protocol: TCP
+      targetPort: 14267
+    selector:
+      name: ${JAEGER_SERVICE_NAME}-collector
+    type: ClusterIP
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      app: jaeger
+      jaeger-infra: query-dc
+  spec:
+    replicas: 1
+    triggers:
+      - type: ConfigChange
+    selector:
+      name: ${JAEGER_SERVICE_NAME}-query
+    strategy:
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: jaeger
+          name: ${JAEGER_SERVICE_NAME}-query
+      spec:
+        containers:
+        - capabilities: {}
+          image: jaegertracing/jaeger-query:${IMAGE_VERSION}
+          name: ${JAEGER_SERVICE_NAME}-query
+          ports:
+          - containerPort: 16686
+            protocol: TCP
+          command:
+          - "/go/bin/query-linux"
+          - "-cassandra.servers=cassandra"
+          - "-cassandra.keyspace=${KEYSPACE}"
+          - "--query.static-files=/go/jaeger-ui/"
+          readinessProbe:
+            httpGet:
+              path: "/"
+              port: 16686
+            initialDelaySeconds: 5
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      app: jaeger
+      jaeger-infra: query-service
+  spec:
+    ports:
+    - name: query-http
+      port: 80
+      protocol: TCP
+      targetPort: 16686
+    selector:
+      name: ${JAEGER_SERVICE_NAME}-query
+    type: ClusterIP
+- apiVersion: v1
+  kind: Route
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-query
+    labels:
+      app: jaeger
+      jaeger-infra: query-route
+  spec:
+    tls:
+      insecureEdgeTerminationPolicy: Allow
+      termination: edge
+    to:
+      kind: Service
+      name: ${JAEGER_SERVICE_NAME}-query
+- apiVersion: extensions/v1beta1
+  kind: DaemonSet
+  metadata:
+    namespace: "${NAMESPACE}"
+    name: ${JAEGER_SERVICE_NAME}-agent
+    labels:
+      app: jaeger
+      jaeger-infra: agent-daemonset
+  spec:
+    template:
+      metadata:
+        labels:
+          jaeger-infra: agent-instance
+      spec:
+        containers:
+        - name: agent-instance
+          image: jaegertracing/jaeger-agent:${IMAGE_VERSION}
+          command:
+          - "/go/bin/agent-linux"
+          - "-collector.host-port=jaeger-collector:14267"
```

Comment:
Why should it? `jaeger-collector` is not a real host, it is the Kubernetes service name, which is the official discovery mechanism in Kubernetes.